File: ruby/spec/actionwords.rb
# encoding: UTF-8
require_relative '../src/coffee_machine'


module Actionwords
  def sut
    @sut ||= CoffeeMachine.new
  end

  def start_the_coffee_machine(lang = 'en')
    sut.start(lang)
  end

  def shutdown_coffee_machine
    sut.stop
  end

  def assert_displayed_message(message)
    expect(sut.message).to eq(message)
  end

  def assert_coffee_served
    expect(sut.coffee_served).to be_truthy
  end

  def assert_no_coffee_is_served
    expect(sut.coffee_served).to be_falsey
  end

  def take_a_coffee
    sut.take_coffee
  end

  def empty_coffee_grounds
    sut.empty_grounds
  end

  def fill_beans
    sut.fill_beans
  end

  def fill_water_tank
    sut.fill_tank
  end

  def take_coffees(coffee_number = 10)
    while (coffee_number > 0)
      take_a_coffee
      coffee_number = coffee_number - 1
    end
  end
end

File: specflow/features/Actionwords.cs
namespace Hiptest.Publisher.Samples {
    using NUnit.Framework;

    public class Actionwords {
        CoffeeMachine Sut = new CoffeeMachine();

        public void StartTheCoffeeMachine(string lang = "en") {
            Sut.Start(lang);
        }

        public void ShutdownCoffeeMachine() {
            Sut.Stop();
        }

        public void AssertDisplayedMessage(string message = "") {
            Assert.AreEqual(message, Sut.Message);
        }

        public void AssertCoffeeServed() {
            Assert.IsTrue(Sut.CoffeeServed);
        }

        public void AssertNoCoffeeIsServed() {
            Assert.IsFalse(Sut.CoffeeServed);
        }

        public void TakeACoffee() {
            Sut.TakeCoffee();
        }

        public void EmptyCoffeeGrounds() {
            Sut.EmptyGrounds();
        }

        public void FillBeans() {
            Sut.FillBeans();
        }

        public void FillWaterTank() {
            Sut.FillTank();
        }

        public void TakeCoffees(int coffeeNumber) {
            while ((coffeeNumber > 0)) {
                this.TakeACoffee();
                coffeeNumber = coffeeNumber - 1;
            }
        }
    }
}